File: Count_the_Digits_That_Divide_a_Number.java
class Solution {
    public int countDigits(int num) {
        if(num >= 1 && num <= 9) return 1;
        int count = 0;
        int nums = num;
        while(num != 0){
            int rem = num % 10;
            if(nums % rem == 0){
                count++;
            }
            num /= 10;
        }
        return count;
    }
}

File: Count_the_Digits_That_Divide_a_Number.cpp
class Solution {
public:
    int countDigits(int num) {
        int count =0;
        int q=num;
        while(q!=0){
            int r=q%10;
            if(num%r==0){
                count++;
            }
            q=q/10;
        }
        return count;
        
    }
};
